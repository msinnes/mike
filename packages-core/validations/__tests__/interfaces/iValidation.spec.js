const { isInterface } = require('@mike/class');

const iValidation = require('../../interfaces/iValidation');

describe('iValidation', () => {
  it('should be an interface', () => {
    expect(isInterface(iValidation)).toBe(true);
  });

  it('should accept a valid config', () => {
    expect(() => {
      iValidation.ensure({
        allowEmpty: true,
        throwOnInvalid: true,
        validateFn: function() {},
      });
    }).not.toThrow();
    expect(() => {
      iValidation.ensure({
        allowEmpty: null,
        throwOnInvalid: null,
        validateFn: function() {},
      });
    }).not.toThrow();
  });

  it('should throw an error if the config is invalid', () => {
    expect(() => {
      iValidation.ensure({});
    }).toThrowErrorMatchingSnapshot();
  });
});
