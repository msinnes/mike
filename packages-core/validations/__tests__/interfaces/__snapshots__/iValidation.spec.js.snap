// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iValidation should throw an error if the config is invalid 1`] = `
"ClassError: Runtime Interface Error
allowEmpty: Expected Boolean
throwOnInvalid: Expected Boolean
validateFn: Expected Function"
`;
